ros2topic/ros2topic/verb/echo.py

```diff
@@ -76,6 +77,11 @@ def add_arguments(self, parser, cli_name):
             '--no-arr', action='store_true', help="Don't print array fields of messages")
         parser.add_argument(
             '--no-str', action='store_true', help="Don't print string fields of messages")
+        parser.add_argument( 
+            '--once', action='store_true', help="Print the first message received and then exit")
+        parser.add_argument(
+            '--timeout', metavar='N', type=unsigned_int, default=3.0,
+            help='If used with --once, the time after which the application will exit even if no message is received')
```

Comment:
@craigh92 why does the timeout only apply when `--once` is used? I can see value in using it for regular, periodic echo calls as well.

Reply:
Good point. I've now changed this.
The `def subscriber()` function now always calls `spin_untill_future_complete()` (I previously had it call the regular `spin()` if `--once` was not used). This means that the timeout can be used even when `--once` is not used. It will however still spin forever (or until a timeout) as the future will never be complete (unless `--once`) is used, so it still behaves the same as it used to.